### DISCUSSION TITLE
Automatically parse when using adapter after successful validation (Zod)

### DISCUSSION THREAD

**Discussion by bennettdams** [TanStack/form#714] [Q&A]
Hey!

Maybe I missed something, but is there a way or plans to automatically parse the form data after a successful validation?

Here's the Zod example, modified to alert the current raw value and parsed value:

https://stackblitz.com/edit/tanstack-form-7lnf2b?file=src%2Findex.tsx&preset=node

If you put in a number like `0123`, you can see that the value in `onSubmit` is still `0123`.
When parsing that value, you'll get `123`.

![image](https://github.com/TanStack/form/assets/29319414/25b14ce6-a329-433e-be50-b2e5239e5bd7)

In other words, the validated data is not parsed with Zod.

PS: The same is true for Zod's `coerce` and `transform`, that's why my example shows both to demonstrate.
